docs/en/rules/Azure.APIM.ProductSubscription.md

```diff
@@ -23,8 +23,69 @@ When combined with policies, subscriptions allow controls such as throttling to
 
 Consider configuring all API Management products to require a subscription.
 
+## EXAMPLES
+
+### Configure with Azure template
+
+To set the display name and the description
+
+set properties.displayName	for the resource type "apis". Dispaly name must be 1 to 300 characters long.
+
+set	properties.description resource type "apis". May include HTML formatting tags.
+
+For example:
+
+```json
+
+{
+  "type": "Microsoft.ApiManagement/service/products",
+  "apiVersion": "2021-12-01-preview",
+  "name": "apim-contoso-test-001/custom-product",
+  "properties": {
+    "approvalRequired": true, 
+    "description": "Custom Product, subscription and approval is required to get the subscription key to call the APIs in Contoso", 
+    "displayName": "Custom Product", 
+    "state": "published",
+    "subscriptionRequired": true, # <----------------- Subscription required enabled
+    "subscriptionsLimit": 1
+  }
+}
+
+```
+
+### Configure with Bicep
+
+To set the display name and the description
+
+set properties.displayName	for the resource "Microsoft.ApiManagement/service/apis@2021-08-01". Dispaly name must be 1 to 300 characters long.
+
+set	properties.description for the resource "Microsoft.ApiManagement/service/apis@2021-08-01". May include HTML formatting tags.
+
+For example:
+
+```bicep
+
+resource api 'Microsoft.ApiManagement/service/apis@2021-08-01' = {
+  parent: service
+  name: 'echo-v1'
+  properties: {
+    displayName: 'Echo API'
+    description: 'An echo API service.'
+    path: 'echo'
+    serviceUrl: 'https://echo.contoso.com'
+    protocols: [
+      'https'
+    ]
+    apiVersion: 'v1'
+    apiVersionSetId: version.id
+    subscriptionRequired: true // <----------------- Subscription required enabled
+  }
```

Comment:
Suggested update to product instead of API.

```suggestion
resource product 'Microsoft.ApiManagement/service/products@2021-08-01' = {
  parent: service
  name: 'echo'
  properties: {
    displayName: 'Echo'
    description: 'Echo API services for Contoso.'
    approvalRequired: true
    subscriptionRequired: true
  }
}
```